compiler/src/scala/quoted/runtime/impl/QuoteMatcher.scala

```diff
@@ -452,20 +433,52 @@ object QuoteMatcher {
       accumulator.apply(Set.empty, term)
   }
 
+  enum MatchResult:
+    case ClosedTree(tree: Tree)
+    case OpenTree(tree: Tree, patternTpe: Type, args: List[Tree], env: Env)
```

Comment:
Woudl be nice to document the expectation on the valid values of the arguments here.